src/tests.rs

```diff
@@ -2553,6 +2553,7 @@ fn cycle() {
 
     // Try to create a cycle: move A into B, and B into the menu, but keep
     // B's parent by children as A.
+    // i.e. folderBBBBBB -> folderAAAAAA -> folderBBBBBB -> menu -> root
     let mut b: Builder = nodes!({ ("menu________", Folder) }).try_into().unwrap();
```

Comment:
Hmm, I think there are a couple of things we can test now:

* Cycles with different depths, like `D -> C -> B -> A -> B -> A -> ...`, in addition to `B -> A -> B -> ...`.
* Cycles with different depths in the same tree—since we used to bail early, but now we'll fix them up.
* A case where a root forms a cycle, like `C -> B -> A -> unfiled -> B -> A -> unfiled -> ...`. `ResolveParent` should break this first by moving `unfiled` to the root.

In those cases, we'll want to make sure we reupload `unfiled` and the reparented folder that starts the cycle.